scripts/run_tests.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+
+JUPYTER_PORT=32323
+IP_ADDRESS="127.0.0.1"
+OUTPUT_FILE=/tmp/output
+run_unit=0
+run_integration=0
+run_docker=0
+CONTAINER=
+
+while getopts 'iuc:h' opt;
+do
+    case "$opt" in
+        c) run_docker=1
+        IP_ADDRESS="0.0.0.0"
+        CONTAINER=${OPTARG}
+        ;;
+
+        i) run_integration=1 ;;
+
+        u) run_unit=1 ;;
+
+        ?|h)
+        echo "Usage: $(basename $0) [-i] [-u] [-c container_name]"
+        echo "-i : run integration tests"
+        echo "-u : run unit tests"
+        echo "-c container_name: run tests against the specified docker container "
+        exit 1
+        ;;
+    esac
+done
+
+# check whether the IP address and port are already in use
+if ( nc -zv $IP_ADDRESS $JUPYTER_PORT 2>&1 >/dev/null ); then
+    echo "The address ${IP_ADDRESS}:${JUPYTER_PORT} is already in use; please kill the existing notebook server to run tests."
+    exit 127
+fi
+
+if [ "$run_docker" == 1 ]; then
+    echo "starting narrative container ${CONTAINER}"
+    rm -f $OUTPUT_FILE
+    touch $OUTPUT_FILE
+    docker run -p $JUPYTER_PORT:$JUPYTER_PORT $CONTAINER \
+        --template \
+        "/kb/dev_container/narrative/src/config.json.templ:/kb/dev_container/narrative/src/config.json" \
+        --template \
+        "/kb/dev_container/narrative/src/config.json.templ:/kb/dev_container/narrative/kbase-extension/static/kbase/config/config.json" \
+        kbase-narrative --no-browser --NotebookApp.allow_origin="*" --ip=$IP_ADDRESS --port=$JUPYTER_PORT 2>&1 | tee $OUTPUT_FILE &
+    bg_pid=$!
+else
+    echo "starting local narrative"
+    rm -f $OUTPUT_FILE
+    touch $OUTPUT_FILE
+    kbase-narrative --no-browser --NotebookApp.allow_origin="*" --ip=$IP_ADDRESS --port=$JUPYTER_PORT 2>&1 | tee $OUTPUT_FILE &
+    bg_pid=$!
+fi
+
+while :; do
```

Comment:
Parsing the output from a docker container was causing problems in the GHA environment, so instead I piped it to a file and then grepped the file for the relevant lines. The docker container outputs STDERR / STDOUT to a file on a shared volume so it can be read by this script.